ibis/expr/types.py

```diff
@@ -529,11 +537,15 @@ def name(self, name):
 
 class ScalarExpr(ValueExpr):
 
-    pass
+    def _type_display(self):
+        return str(self.type())
 
 
 class ArrayExpr(ValueExpr):
 
+    def _type_display(self):
+        return 'array({0!s})'.format(self.type())
```

Comment:
Unless we're supporting 2.6 this can just be `{}`. Small nit, and no need to change if you don't want to.